elyra/pipeline/airflow/component_parser_airflow.py

```diff
@@ -46,149 +40,367 @@ class AirflowComponentParser(ComponentParser):
+                'docstring': self._get_class_docstring(operator) or ''
+            } for operator in operator_classes
         }
 
-        class_name = "no_class"
-        class_regex = re.compile(r"class ([\w]+[.\w]*)\(\w*\):")
-        for line in component_definition.split('\n'):
-            # Remove any inline comments (must follow the '2 preceding spaces and one following space'
-            # rule). This avoids the case where the default value of an __init__ arg contains '#'.
-            line = re.sub(r"  # .*\n?", "", line)
-            match = class_regex.search(line)
-            if match:
-                class_name = match.group(1)
-                class_to_content[class_name] = {"lines": [], "args": []}
-            class_to_content[class_name]['lines'].append(line)
-
-        class_to_content.pop("no_class")
-
-        init_regex = re.compile(r"def __init__\(([\s\d\w,=\-\'\"\[\]\{\}\*\s\#.\\\/:?]*)\):")
-        for class_name, content in class_to_content.items():
-            # Concatenate class body and search for __init__ function
-            class_content = '\n'.join(content['lines'])
-            # remove type hint via TypeHintRemover with ast parse
-            parsed_source = ast.parse(class_content)
-            transformed = TypeHintRemover().visit(parsed_source)
-            class_content = astunparse.unparse(transformed)
-
-            for match in init_regex.finditer(class_content):
-                # Get list of parameter:default-value pairs
-                class_to_content[class_name]['args'] = [x.strip() for x in match.group(1).split(',') if x.strip() != '']
-
-        return class_to_content
-
-    def _parse_properties(self, content: Dict[str, List]) -> List[ComponentParameter]:
-        properties: List[ComponentParameter] = list()
+    def _filter_operator_classes(self,
+                                 class_def_nodes: List[ast.ClassDef],
+                                 import_from_nodes: List[ast.ImportFrom]) -> List[ast.ClassDef]:
+        """
+        Analyze each ast.ClassDef object to determine whether it directly or indirectly
+        extends the BaseOperator.
+
+        :param class_def_nodes: a list of all ast.ClassDef objects in a file
+        :param import_from_nodes: a list of all ast.ImportFrom objects in a file (only used to satisfy
+                                  the temporary workaround called out below)
+
+        :returns: a filtered list of ast.ClassDef objects that can be considered 'Operators'
+        """
+        operator_classes = []
+        classes_to_analyze = []
+
+        # Get class names for package imports that match one of the following patterns,
+        # indicating that this class does match a known Operator class as defined in
+        # a provider package or core Airflow package
+        regex_patterns = [
+            re.compile(r'airflow\.providers\.[a-z_]+\.operators'),  # airflow.providers.*.operators. (provider package)
```

Comment:
Should this regex include upper-case as well as digits or does airflow restrict provider names to lower-case alpha?

Reply:
Hmm good question. I'll look into it

Reply:
I think we're safe sticking to all lower case since we're looking for the imported modules in this case ([justification](https://www.python.org/dev/peps/pep-0008/#package-and-module-names))

Reply:
That's a fair point.  

I think digits should be included though.  Here are some of the packages in my current elyra env with digits: argon2, flake8, httplib2, jinja2, json5, s3contents,  urllib3

Hmm.  I also see two packages with uppercased characters: `Cryptodome` and `IPython` and here's an[ import statement referencing the latter](https://github.com/ipython/ipykernel/blob/6ea9018f9f7e6554181c52ef9610787c2ca05548/ipykernel/kernelbase.py#L37).  I'm sure this is rare, not to mention unconventional, but it seems like we'd be safer with:
```suggestion
            re.compile(r'airflow\.providers\.[a-zA-Z0-9_]+\.operators'),  # airflow.providers.*.operators. (provider package)
```

Reply:
Fair enough!